2020/day21/solution.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import annotations
+from collections import Counter
+import re
+
+food_re = re.compile(r"(?P<ingredients>.*) \(contains (?P<allergens>.*)\)")
+
+def get_ingredients_and_allergens(line: str) -> tuple[set[str], set[str]]:
+    match = food_re.match(line)
+    ingredients = set(ingredient.strip() for ingredient in match['ingredients'].split(' '))
+    allergens = set(ingredient.strip() for ingredient in match['allergens'].split(','))
```

Comment:
```suggestion
    ingredients = set(ingredient for ingredient in match['ingredients'].split(' '))
    allergens = set(allergen for allergen in match['allergens'].split(', '))
```
Didn't actually check but seems like you can get rid of the `strip()s` with `','` -> `', '`?

Reply:
It does.  Good catch